plugins/repository-s3/build.gradle

```diff
@@ -495,3 +534,28 @@ if (BuildParams.runtimeJavaVersion <= JavaVersion.VERSION_1_8) {
 } else {
   thirdPartyAudit.ignoreMissingClasses 'javax.activation.DataHandler'
 }
+
+["test", "internalClusterTest", "s3ThirdPartyTest", "testRepositoryCreds"].forEach { action ->
+  tasks.named(action).configure {
+    dependsOn "writeTestJavaPolicy"
+    systemProperty 'java.security.policy', "file://${buildDir}/tmp/java.policy"
+  }
+}
+
+tasks.register("writeTestJavaPolicy") {
+  doLast {
+    final File tmp = file("${buildDir}/tmp")
+    if (tmp.exists() == false && tmp.mkdirs() == false) {
+      throw new GradleException("failed to create temporary directory [${tmp}]")
+    }
+    final File javaPolicy = file("${tmp}/java.policy")
+    javaPolicy.write(
+      [
+        "grant {",
```

Comment:
I believe we could just create a `test.policy` in src/test/resources (this is static file anyway) and just set it here https://github.com/opensearch-project/OpenSearch/pull/7372/files#diff-7c9c630a816f820ccc1b9f6d801e7163cea5031c22cab6178d689ae3ec606ec3R541

Reply:
For repository-s3, that is how we were setting permissions for the test environment, but discovery-ec2 seems to be using both, the security policy file as well as the gradle target, not sure why.